### DISCUSSION TITLE
resource consumption

### DISCUSSION THREAD

**Discussion by flixman** [goauthentik/authentik#9569] [Q&A]
Hi everybody! I have been using another identity provider, keycloak, for a while, and I'd like to switch to authentik. There is something that particularly annoys me, that is: even with a just bunch of users (10-20), keycloak takes easily 400 MB of RAM. What would be the correspondent memory consumption of authentic for such a small use case?

**Comment by codespearhead** [comment]
Half a gigabyte of RAM is the bare minimum for any production application these days.

Take a look at PaaS offerings such as [AWS Lightsail (container offering)](https://aws.amazon.com/pt/lightsail/pricing/), [Heroku (Eco plan)](https://www.heroku.com/pricing) and [Digital Ocean (App Platform)](https://www.digitalocean.com/pricing/app-platform): none of them has a plan with less than 512MB of RAM.

[Zitadel](https://zitadel.com/docs/self-hosting/manage/production#general-resource-usage) and [Authentik are no different](https://docs.goauthentik.io/docs/installation/docker-compose#requirements) (unofficial benchmark [here](https://www.reddit.com/r/selfhosted/comments/14dl23q/comment/jos3ggp/#:~:text=authentik_server)).

**Reply by helmut72** [reply]
Zitadel use a lot less RAM in practice with just a few users (under 150MB), also Authelia stays under 100MB RAM. Both don't use Java or Python.

**Reply by flixman** [reply]
I have seriously encountered feelings on this matter: Most of today's projects are based on frameworks that simplify the development process, but then are not optimized for their deployment. This way, one ends up requiring 1 GB of RAM for a messaging app, or 512 MB for authentication servers. Taking as an example an openldap server, for the same amount of users it shows a consumption of 40-50 MB: 10x less. I like both keycloak and authentik, but in general... the baseline resource consumption is through the roof (as in any other app developed these days).

In any case, thank you very much for your answers!
